src/sqlancer/ComparatorHelper.java

```diff
@@ -87,13 +87,19 @@ public static List<String> getResultSetFirstColumnAsString(String queryString, E
     public static void assumeResultSetsAreEqual(List<String> resultSet, List<String> secondResultSet,
             String originalQueryString, List<String> combinedString, SQLGlobalState<?, ?> state) {
         if (resultSet.size() != secondResultSet.size()) {
-            String queryFormatString = "-- %s;\n-- cardinality: %d";
+            String queryFormatString = "-- %s;" + System.lineSeparator() + "-- cardinality: %d"
+                    + System.lineSeparator();
             String firstQueryString = String.format(queryFormatString, originalQueryString, resultSet.size());
-            String secondQueryString = String.format(queryFormatString,
-                    combinedString.stream().collect(Collectors.joining(";")), secondResultSet.size());
-            state.getState().getLocalState().log(String.format("%s\n%s", firstQueryString, secondQueryString));
-            String assertionMessage = String.format("the size of the result sets mismatch (%d and %d)!\n%s\n%s",
-                    resultSet.size(), secondResultSet.size(), firstQueryString, secondQueryString);
+            String combinedQueryString = String.join(";", combinedString);
+            String secondQueryString = String.format(queryFormatString, combinedQueryString, secondResultSet.size());
+            state.getState().getLocalState()
+                    .log(String.format("%s" + System.lineSeparator() + "%s", firstQueryString, secondQueryString));
+            String assertionMessage = String.format(
+                    "the size of the result sets mismatch (%d and %d)!" + System.lineSeparator()
+                            + "FirstQuery: \"%s\", whose cardinality is: %d" + System.lineSeparator()
```

Comment:
Since these are not variable names, I think we could just use "first query: " and "second query: "?

Reply:
make sense, done.